### DISCUSSION TITLE
Working on advanced splitter

### DISCUSSION THREAD

**Discussion by LexiestLeszek** [ZachNagengast/similarity-search-kit#15] [Ideas]
Hi! NaturalLanguage swift library has built-in tokenizers for .paragraph / .sentence / .word 

https://developer.apple.com/documentation/naturallanguage/tokenizing_natural_language_text
https://developer.apple.com/documentation/naturallanguage/nltokenunit/paragraph
https://developer.apple.com/documentation/naturallanguage/nltokenunit/sentence
https://developer.apple.com/documentation/naturallanguage/nltokenunit/word

Does it make sense to use it for text splitting instead?

**Comment by ZachNagengast** [comment]
I have the word one in the library currently, it would be interesting to see the use of sentences/paragraphs but I can't think of a use case personally.
